tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/DirectSubmission/LogFactoryGetConfigurationForLoggerInstrumentation.cs

```diff
@@ -35,6 +41,12 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmi
     [EditorBrowsable(EditorBrowsableState.Never)]
     public class LogFactoryGetConfigurationForLoggerInstrumentation
     {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LogFactoryGetConfigurationForLoggerInstrumentation));
+        private static NLogVersion? _version;
+
+        // TODO attempt to store the fact that we've configured a given configuration to not double configure
+        internal static ConditionalWeakTable<object, object> ConfigurationTable { get; } = new();
```

Comment:
So testing it out again, yes the only issue that I was attempting to avoid with this was the double direct log submission.

But it got me wondering why it was an issue at all and it looks like for pre 4.3 NLog the check for protecting against adding another direct log submission target was for the wrong type. So I fixed that and even without this `ConditionalWeakTable` it now works as expected.

Thoughts on just removing this `ConfigurationTable` now? It'd then fall through and be handled correctly now when checking whether the direct submission target should be added.